src/core/lib/event_engine/channel_args_endpoint_config.h

```diff
@@ -16,27 +16,35 @@
 
 #include <grpc/support/port_platform.h>
 
+#include <memory>
+
 #include "absl/strings/string_view.h"
 
 #include <grpc/event_engine/endpoint_config.h>
 #include <grpc/impl/codegen/grpc_types.h>
 
+#include "src/core/lib/channel/channel_args.h"
+
 namespace grpc_event_engine {
 namespace experimental {
 
-/// A readonly \a EndpointConfig based on grpc_channel_args. This class does not
-/// take ownership of the grpc_endpoint_args*, and instances of this class
-/// should not be used after the underlying args are destroyed.
 class ChannelArgsEndpointConfig : public EndpointConfig {
  public:
-  explicit ChannelArgsEndpointConfig(const grpc_channel_args* args)
+  explicit ChannelArgsEndpointConfig(const grpc_core::ChannelArgs& args)
       : args_(args) {}
-  Setting Get(absl::string_view key) const override;
+  explicit ChannelArgsEndpointConfig(const grpc_channel_args* args)
+      : args_(grpc_core::ChannelArgs::FromC(args)) {}
+  EndpointConfig::Setting Get(absl::string_view key) const override;
 
  private:
-  const grpc_channel_args* args_;
+  grpc_core::ChannelArgs args_;
 };
 
+std::unique_ptr<EndpointConfig> CreateEndpointConfig(
```

Comment:
These factory functions don't seem to add much value, and they force the caller to use dynamic allocation for no real benefit.  Instead, I think callers can just instantiate `ChannelArgsEndpointConfig` directly.  For example, the chttp2 client code can do this:

```
ChannelArgsEndpointConfig config(final_args);
grpc_endpoint* client = grpc_tcp_create_from_fd(
    grpc_fd_create(fd, "client", true), config, "fd-client");
```

Note that `ChannelArgs` is copyable and movable, so there's no reason `ChannelArgsEndpointConfig` can't be as well.

Reply:
Removed them.